storage/dataflux/fast_list.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataflux
+
+import (
+	"context"
+
+	"cloud.google.com/go/storage"
+)
+
+// listingMethod represents the method of listing.
+type listingMethod int
+
+const (
+	// open when any method can be used to list.
+	open listingMethod = iota
+	// sequential when the listing is done sequentially.
+	sequential
+	// worksteal when the listing is done using work stealing algorithm.
+	worksteal
+)
+
+// Input contains options for listing objects.
+type Input struct {
+	// Client is the storage client to list objects from.
+	Client *storage.Client
+	// BucketName is the name of the bucket to list objects from.
+	BucketName string
+	// Parallelism is number of parallel workers to use for listing.
+	Parallelism int
+	// pageSize is the number of objects to list.
+	BatchSize int
+	// query is the query to filter objects for listing.
```

Comment:
Capitalize Query to match the param name.

Reply:
Done.